eth/backend.go

```diff
@@ -330,6 +330,38 @@ func (s *Ethereum) Etherbase() (eb common.Address, err error) {
 	return common.Address{}, fmt.Errorf("etherbase must be explicitly specified")
 }
 
+// isLocalAccount checks whether the specified address is a local account.
+//
+// This function is used during block chain reorg checking to determine
+// whether a block is mined by local accounts. We regard three types of
+// accounts as local account: etherbase, accounts in keystore or accounts
+// specified via `txpool.locals` flag.
+func (s *Ethereum) isLocalAccount(addr common.Address) bool {
+	s.lock.RLock()
+	etherbase := s.etherbase
+	s.lock.RUnlock()
+	// Check whether the given address is etherbase.
```

Comment:
Pleas move this comment to the top before the locking. That code segment is also already part of the check.